### PR TITLE
Hoist all common packages to the workspace root

### PR DESCRIPTION
**Pull request by steveluscher**
This should make it easier to upgrade things like TypeScript and the linter configs.

> [!NOTE]
> I had to do some terrible things to keep `@types/jest` out of the way of `@types/mocha` in the legacy library.

### CODE REVIEW

**Comment by steveluscher** [comment]
* **#2272** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2272?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2271** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2271?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @steveluscher and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/accounts/package.json [hunk 1]
```diff
@@ -72,24 +72,9 @@
...
-        "eslint": "^8.45.0",
-        "eslint-plugin-sort-keys-fix": "^1.1.2",
-        "jest": "^29.7.0",
-        "jest-runner-eslint": "^2.2.0",
-        "jest-runner-prettier": "^1.0.0",
-        "prettier": "^3.1",
-        "tsup": "^8.0.1",
-        "typescript": "^5.2.2",
-        "version-from-git": "^1.1.1"
+        "@solana/tsconfig": "workspace:*"
```
Comment (buffalojoec) [inline packages/accounts/package.json:77]: Can these `workspace:*` ones get hoisted, too?

File: packages/library-legacy/tsconfig.d.json [hunk 2]
```diff
@@ -2,7 +2,14 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "emitDeclarationOnly": true,
-    "stripInternal": true
+    "stripInternal": true,
+    "paths": {
+      "@types/jest": [
+        "./test/__shadow-jest-types.d.ts"
+      ]
+    }
```
Comment (buffalojoec) [inline packages/library-legacy/tsconfig.d.json:10]: Doesn't look all that terrible!

**Review by buffalojoec — APPROVED** [review]
hoist

File: packages/accounts/package.json (hunk 1 above)
Comment (steveluscher) [reply packages/accounts/package.json:77]: Maybe. I'm afraid.

**Comment by steveluscher** [comment]
### Merge activity

* **Mar 6, 5:57 PM PST**: @steveluscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2271).

File: packages/accounts/package.json [hunk 3]
```diff
@@ -34,20 +34,20 @@
...
-        "dev": "jest -c node_modules/@solana/test-config/jest-dev.config.ts --rootDir . --watch",
+        "compile:typedefs": "tsc -p ./tsconfig.declarations.json && node ../../node_modules/@solana/build-scripts/add-js-extension-to-types.mjs",
+        "dev": "jest -c ../../node_modules/@solana/test-config/jest-dev.config.ts --rootDir . --watch",
         "prepublishOnly": "version-from-git --no-git-tag-version --template experimental.short && pnpm pkg delete devDependencies",
         "publish-impl": "npm view $npm_package_name@$npm_package_version > /dev/null 2>&1 || pnpm publish --tag experimental --access public --no-git-checks",
         "publish-packages": "pnpm prepublishOnly && pnpm publish-impl",
         "style:fix": "pnpm eslint --fix src/* && pnpm prettier -w src/* package.json",
-        "test:lint": "jest -c node_modules/@solana/test-config/jest-lint.config.ts --rootDir . --silent",
-        "test:prettier": "jest -c node_modules/@solana/test-config/jest-prettier.config.ts --rootDir . --silent",
+        "test:lint": "jest -c ../../node_modules/@solana/test-config/jest-lint.config.ts --rootDir . --silent",
```
Comment (buffalojoec) [inline packages/accounts/package.json:43]: Ahh yuck. Shame there's not like a `$WORKSPACE_ROOT` variable or something.

**Pull request merged**